ReactAndroid/src/main/java/com/facebook/react/views/text/ReactTextShadowNode.java

```diff
@@ -439,8 +452,18 @@ public void setTextAlign(@Nullable String textAlign) {
 
   @ReactProp(name = ViewProps.FONT_SIZE, defaultFloat = UNSET)
   public void setFontSize(float fontSize) {
-    if (fontSize != UNSET) {
-      fontSize = (float) Math.ceil(PixelUtil.toPixelFromSP(fontSize));
+    setFontSize(fontSize, false);
+  }
+
+  public void setFontSize(float fontSize, boolean forceRecalculation) {
```

Comment:
What's the purpose of the `forceRecalculation` flag? It seems to only be used by `setAllowFontScaling` and `setAllowFontScaling` only calls `setFontSize` under conditions such that whole function would run even if `forceRecalculation` was `false`.

I'm thinking we should get rid of the `forceRecalculation` flag and have `setAllowFontScaling` always call `setFontSize`. I think this would make the code more straight forward, it would behave the same way, and it wouldn't introduce any noticeable performance overhead. What do you think?